### ISSUE TITLE
poloniex websocket

### ISSUE THREAD

**Issue by p0nt** [Aurora-Team/BitcoinExchangeFH#92]
poloniex has a websocket.

https://poloniex.com/support/api/

**Comment by gavincyi** [comment]
Yes the reason is the dependence is on WAMP protocol while I still find anyone (neither exchange nor third-party) wrote a websocket connector at the moment I developed it. In my understanding the easiest way is to employ [autobahn-python](https://github.com/crossbario/autobahn-python) and it still takes some time. I don't think I have time to develop a websocket one in a short while unless I have an urgent need (short term trading).

**Comment by Affliction123** [comment]
@gavincyi I've been using a solution with websocket that writes to MySQL from Poloniex, would You be interested to use it in a project? I could make a pull request or you can check it out, its quite straight forward and works well. I subscribe to BTC_ETH, and add timestamps as they are not added in the stream in form of unix epoch and %Y-%m-%d %H:%M:%S.%f, check if message is a trade or an order, and insert each appropriately.

    import websocket
    import pymysql.cursors
    import json
    import time
    from datetime import datetime
    import pandas as pd

    connection = pymysql.connect(host='localhost',
                             user='X',
                             password='Y',
                             db='Z',
                             charset='utf8mb4',
                             cursorclass=pymysql.cursors.DictCursor)

    def on_error(ws, error):
        print(error)

    def on_close(ws):
        print("### closed ###")
        connection.close()

    def on_open(ws):
        print("ONOPEN")
        ws.send(json.dumps({'command':'subscribe','channel':'BTC_ETH'}))

    def on_message(ws, message):
        d1 = time.time()
        d2 = datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")
        message = json.loads(message)
        if message[2]:
            seq = int(message[1])
            for k in message[2]:
                if len(k) == 4:
                    try:
                        with connection.cursor() as cursor:
                            sql = "INSERT INTO `zerozerozero` (`typeOT`, `typeSB`, `rate`, `amount`,`seq`, `timeDateOTe`, `timeDateOTh`) VALUES (%s, %s, %s, %s, %s, %s, %s)"
                            cursor.execute(sql, (k[0], k[1], k[2], k[3], seq, d1, d2))
                            connection.commit()
                            cursor.close()
                    finally:
                        pass

                elif len(k) == 6:
                    try:
                        with connection.cursor() as cursor:
                            sql1 = "INSERT INTO `zerozerozero` (`typeOT`, `tradeID`, `typeSB`, `rate`, `amount`, `timeDateT`,`seq`, `timeDateOTe`, `timeDateOTh`) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)"
                            d3 = int(k[5])
                            cursor.execute(sql1, (k[0], k[1], k[2], k[3], k[4], d3, seq, d1, d2))
                            connection.commit()
                            cursor.close()
                    finally:
                        pass
                
    websocket.enableTrace(True)
    ws = websocket.WebSocketApp("wss://api2.poloniex.com/",
                              on_message = on_message,
                              on_error = on_error,
                              on_close = on_close)
    ws.on_open = on_open
    ws.run_forever()

**Comment by enalco** [comment]
Hey, [Poloniex ws codes](https://gist.github.com/enalco/e8345d0d7c75a7e48d32dfb4df30e237) probably help
